src/api/index.js

```diff
@@ -0,0 +1,272 @@
+            token: localStorage.token
+
+        })
+    };
+
+    fetchDict = () => {
+        return this.makeRequest({
+            url: `${this.url}/user/dict`,
+            data: {
+                method: 'GET'
+            },
+            token: localStorage.token
+
+        })
+    };
+
+    fetchLanguages = () => {
+        return this.makeRequest({
+            url: `${this.url}/languages`,
+            data: {
+                method: 'GET'
+            },
+            token: localStorage.token
+
+        })
+    };
+
+    fetchDeleteWord = (id) => {
+        return this.makeRequest({
+            url: `${this.url}/user/dict?word_id=${id}`,
+            data: {
+                method: 'DELETE'
+            },
+            token: localStorage.token
+
+        })
+    };
+
+
+    fetchLogOutUser = () => {
+        return this.makeRequest({
+            url: `${this.url}/logout`,
+            data: {
+                method: 'GET'
+            },
+        })
+    };
+
+    fetchSongs = (page, limit) => {
+        return this.makeRequest(
+            {
+                url: `${this.url}/songs?page=${page}&limit=${limit}`,
+                data: {
+                    method: 'GET'
+                },
+                token: localStorage.token
+
+            })
+    };
+
+    fetchLikeSong = id => {
+        return this.makeRequest({
+            url: `${this.url}/songs/${id}/like`,
+            data: {
+                method: 'POST',
+            },
+            token: localStorage.token
+
+        })
+    };
+    fetchFindSong = (title, artist) => {
+        return this.makeRequest({
+            url: `${this.url}/songs/search?title=${title}&artist=${artist}`,
+            data: {
+                method: 'GET',
+
+            },
+            token: localStorage.token
+
+        })
+    };
+    fetchLikedSongs = () => {
+        return this.makeRequest({
+            url: `${this.url}/songs/liked`,
+            data: {
+                method: 'GET'
+            },
+            token: localStorage.token
+
+        })
+    };
+    fetchSongsByArtist = (artist) => {
+        return this.makeRequest({
+            url: `${this.url}/songs?artist=${artist}`,
+            data: {
+                method: 'GET'
+            },
+            token: localStorage.token
+        })
+    };
+    fetchTranslateWord = (word) => {
+        return this.makeRequest({
+            url: `${this.url}/words/translate?word=${word}`,
+            data: {
+                method: 'GET'
+            },
+            token: localStorage.token
+        })
+    };
+    fetchDataUser = () => {
+        return this.makeRequest({
+            url: `${this.url}/user`,
+            data: {
+                method: 'GET'
+            },
+            token: localStorage.token
+        })
+    };
+    fetchUserUpdateData = (data) => {
+        return this.makeRequest({
+            url: `${this.url}/user`,
+            data: {
+                method: 'PATCH',
```

Comment:
PATCH method according to REST API is refered to cases wheb you partially upading object and if updating whole object there is used PUT method.
Please take a look to this

Reply:
yep, I update not a whole user